autoload/intero/util.vim

```diff
@@ -69,4 +73,113 @@ function! intero#util#tocol(line, col) abort "{{{
     return l:len + 1
 endfunction "}}}
 
+function! s:load_targets_from_stack() abort
+    return systemlist('stack ide targets')
+endfunction
+
+if (!exists('g:intero_load_targets'))
+    " A list of load targets.
+    let g:intero_load_targets = []
+endif
+
+" Attempt to set the load targets. When passed an empty array, this uses the
+" targets as given by `stack ide targets`.
+function! intero#util#set_load_targets(targets) abort
+    if len(a:targets) == 0
+        let g:intero_load_targets = s:load_targets_from_stack()
+        return g:intero_load_targets
+    endif
+
+    " if stack targets are empty, then we are not in a stack project.
+    " attempting to set the targets will cause the build command to fail.
+    let l:stack_targets = s:load_targets_from_stack()
+    if empty(l:stack_targets)
+        let g:intero_load_targets = []
+        return g:intero_load_targets
+    endif
+
+    " we are in a stack project, and there are desired targets. validate that
+    " they are contained inside the stack load targets
+""    for target in a:targets
+""        if index(l:stack_targets, target) == -1
+""            call intero#util#print_warning("Target " . target . " not present in available Stack targets: " . join(l:stack_targets, ' '))
+""        endif
+""    endfor
+
+    call s:multiple_options(g:intero_load_targets, l:stack_targets)
```

Comment:
Agreed; this is code that was adapted from stack overflow and I'm not sure how to use vim's control flow mechanism's here.